### PR TITLE
Create a Pull Request template

### PR DESCRIPTION
**Pull request by nannanli**
Signed-off-by: nannanli <nannanli@cn.ibm.com>

This PR defines a template for the Pull Requests. It will require submitter to fill in some information to help us automate some processes like the release notes generation.

### CODE REVIEW

**Comment by nannanli** [comment]
Hi @jackjia-ibm @Joe-Winchester @stevenhorsman @OnnoVdT , could you please review this PR template? This is for the `zowe-install-packaging` repo only now to gather early feedback. For the SMP/E hold data, does the description work in the template? 

Thanks!

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+- [ ] DCO signoffs have been added to all commits, including this PR
+
+#### PR type
+What type of changes does your PR introduce to Zowe? _Put an `x` in the box that applies to this PR. If you're unsure about any of them, don't hesitate to ask._ 
+- [ ] Bugfix <!-- non-breaking change which fixes an issue -->
+- [ ] Feature <!-- non-breaking change which adds functionality -->
+- [ ] Code style update (formatting, local variables)
+- [ ] Refactoring (no functional changes, no API changes)
+- [ ] Build related changes
+- [ ] CI related changes
```
Comment (Joe-Winchester) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md]: Do we need `Code style update` , `Build related changes` and `CI related changes` ? in the PR ?   For an end user I am not sure of the value of these being called out.

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
+<!-- Thank you for your pull request! Please provide a description of the changes in this PR in the field above and provide the following information. -->
+
+Please check if your PR fulfills the following requirements. This is simply a reminder of what we are going to look for before merging your PR.
```
Comment (Joe-Winchester) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md]: Add `If you don't know all of this information when you create your PR don't worry - you can edit this template as you're working on it`

Comment (nannanli) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md]: Done.

**Comment by OnnoVdT** [comment]
most of my sample got stripped on submit :( Retryuing here.
**Does this PR do something the person installing Zowe should know about?**

Is this a fix or a feature that changes customizable files (e.g. configuration files, sample JCL, etc), 
product requirements, or other installation or configuration related area? If so, please fill out the template below. There is no need to report the need to restart the servers to activate the change. Note that the provided text will be formatted in 64-character lines, and that round brackets, ( and ), must always be used in matching pairs.
****************************************************************
* Affected function: _general area of interest_                *
****************************************************************
* Description: _1 line description_                            *
****************************************************************
* Part: _name of customizable file involved_                   *
****************************************************************
_multi-line description_

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md (hunk 1 above)
Comment (nannanli) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md]: > Do we need `Code style update` , `` Build related changes``Refactoring (no functional... `` and `CI related changes` ? in the PR ? For an end user I am not sure of the value of these being called out. If anyone wanted to get this information they could do a compare of the files in the PR to determine and having these questions asked may be too much information to capture.

@Joe-Winchester Done. Removed the extra options.

**Comment by nannanli** [comment]
> Hold info header should be boxed in so it's easy to see the individual sections once they are all strung together. Hold info is also missing the "**multiline description**" marker.

Added the missing details now.

**Pull request merged**